modules/meshrenderinggl/include/modules/meshrenderinggl/processors/meshrasterizer.h

```diff
@@ -1,306 +1,315 @@
+ *       + __Color Source__ The source of the color: vertex color, transfer function, or external
+ * constant color
+ *       + __Separate Uniform Alpha__ Overwrite alpha settings from above with a constant alpha
+ * value
+ *       + __Normal Source__ Source of the pixel normal: interpolated or not
+ *       + __Shading Mode__ The shading that is applied to the pixel color
+ *       + __Show Edges__ Show triangle edges
+ *       + __Edge Color__ The color of the edges
+ *       + __Edge Opacity__ Blending of the edge color:
+ *           0-1: blending factor of the edge color into the triangle color, alpha unmodified;
+ *           1-2: full edge color and alpha is increased to fully opaque
+ */
+
+/**
+ * \class MeshRasterizer
+ * \brief Mesh Renderer specialized for rendering highly layered and transparent surfaces.
+ *
+ * Its settings will be used to add fragments to a FragmentListRenderer for the rendering of the
+ * transparent mesh. Many alpha computation modes, shading modes, color modes can be combined and
+ * even selected individually for the front- and back face.
+ */
+class IVW_MODULE_MESHRENDERINGGL_API MeshRasterizer : public Processor {
+    friend class MeshRasterization;
+
+public:
+    MeshRasterizer();
+    virtual ~MeshRasterizer() = default;
+
+    virtual const ProcessorInfo getProcessorInfo() const override;
+    static const ProcessorInfo processorInfo_;
+
+    /**
+     * \brief Outputs a Rasterization instance to be rendered later.
+     */
+    virtual void process() override;
+    virtual void initializeResources() override;
+
+protected:
+    /**
+     * \brief Update the mesh drawer.
+     * This is called when the inport is changed or when a property requires preprocessing steps on
+     * the mesh, e.g. for silhouettes or special alpha features.
+     */
+    void updateMeshes();
+
+    MeshFlatMultiInport inport_;
+    RasterizationOutport outport_;
+
+    CameraProperty camera_;
+    CameraTrackball trackball_;
+    SimpleLightingProperty lightingProperty_;
+
+    BoolProperty forceOpaque_;
+
+    BoolProperty drawSilhouette_;
+    FloatVec4Property silhouetteColor_;
+
+    enum class NormalSource : int { InputVertex, GenerateVertex, GenerateTriangle };
+    TemplateOptionProperty<NormalSource> normalSource_;
+    TemplateOptionProperty<meshutil::CalculateMeshNormalsMode> normalComputationMode_;
+
+    /**
+     * \brief Settings to assemble the equation for the alpha values.
+     * All individual factors are clamped to [0,1].
+     */
+    struct AlphaSettings : public CompositeProperty {
+        AlphaSettings();
+
+        BoolProperty enableUniform_;
+        FloatProperty uniformScaling_;
+        // IRIS
+        BoolProperty enableAngleBased_;
+        FloatProperty angleBasedExponent_;
+        BoolProperty enableNormalVariation_;
+        FloatProperty normalVariationExponent_;
+        // Smoke surfaces
+        BoolProperty enableDensity_;
+        FloatProperty baseDensity_;  // k in the paper

```

Comment:
Which paper? 
Can state that in the AlphaSettings doc?

Reply:
Now citing both papers in the top, since they are mentioned repeatedly